meerkat_frontend/src/js/technical/tabletSubmission.js

```diff
@@ -9,114 +9,106 @@ function gatherTabletData(locID) {
-
-    var month_Api1 = '/devices/submissions/tot_1?location=' + locID + '&filter=date:ge:' + date_builder(1, "month", true) + '&filter=date:le:' + date_builder(1, "month", false);
-    var month_Api = '/devices/submissions/tot_1?location=' + locID + '&filter=date:ge:' + date_builder(0, "month", true) + '&filter=date:le:' + date_builder(0, "month", false);
-
+    var week_tot_Api2 = '/devices/submissions/tot_1?location=' + locID + '&filter=epi_year:eq:' + current_Year + '&filter=epi_week:eq:' + week2;
+
+    apiObjArray.push({
+        "api": '/devices/submissions/tot_1?location=' + locID + '&filter=epi_year:eq:' + current_Year + '&filter=epi_week:eq:' + week1,
+        "minusNum": "1",
+        "columnType": "casesInWeek"
+    }, {
+        "api": '/devices/submissions/tot_1?location=' + locID + '&filter=epi_year:eq:' + current_Year + '&filter=epi_week:eq:' + week,
+        "minusNum": "0",
+        "columnType": "casesInWeek"
+    }, {
+        "api": '/devices/submissions/reg_1?location=' + locID + '&filter=epi_year:eq:' + current_Year + '&filter=epi_week:eq:' + week2,
+        "minusNum": "2",
+        "columnType": "noOfRegistersInWeek"
+    }, {
+        "api": '/devices/submissions/reg_1?location=' + locID + '&filter=epi_year:eq:' + current_Year + '&filter=epi_week:eq:' + week1,
+        "minusNum": "1",
+        "columnType": "noOfRegistersInWeek"
+    }, {
+        "api": '/devices/submissions/reg_1?location=' + locID + '&filter=epi_year:eq:' + current_Year + '&filter=epi_week:eq:' + week,
+        "minusNum": "0",
+        "columnType": "noOfRegistersInWeek"
+    }, {
+        "api": '/devices/submissions/tot_1?location=' + locID + '&filter=date:ge:' + date_builder(2, "year", true) + '&filter=date:le:' + date_builder(2, "year", false),
+        "minusNum": "2",
+        "columnType": "casesInYear"
+    }, {
+        "api": '/devices/submissions/tot_1?location=' + locID + '&filter=date:ge:' + date_builder(2, "year", true) + '&filter=date:le:' + date_builder(1, "year", false),
+        "minusNum": "1",
+        "columnType": "casesInYear"
+    }, {
+        "api": '/devices/submissions/tot_1?location=' + locID + '&filter=date:ge:' + date_builder(2, "year", true) + '&filter=date:le:' + date_builder(0, "year", false),
+        "minusNum": "0",
+        "columnType": "casesInYear"
+    }, {
+        "api": '/devices/submissions/tot_1?location=' + locID + '&filter=date:ge:' + date_builder(1, "month", true) + '&filter=date:le:' + date_builder(1, "month", false),
+        "minusNum": "1",
+        "columnType": "casesInMonthL"
+    }, {
+        "api": '/devices/submissions/tot_1?location=' + locID + '&filter=date:ge:' + date_builder(1, "month", true) + '&filter=date:le:' + date_builder(0, "month", false),
+        "minusNum": "0",
+        "columnType": "casesInMonthC"
+    });
 
 
-    //Start week tot data
+    var deferreds = [];
+    // The first call will generat the most of the table structure
```

Comment:
In my opinion this comment doesn't say much to help understand what's going on.
I think you can safety remove it.

Reply:
edited